src/Executables/Game/ShopsManager.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.Extensions.Logging;
+using QuantumCore.API;
+using QuantumCore.API.Core.Models;
+using QuantumCore.Game.Persistence;
+
+namespace QuantumCore.Game;
+
+public interface IShopsManager
+{
+    public Task<ShopItem[]> GetShopItems(uint npcVnum);
+}
+
+public class ShopsManager : IShopsManager
+{
+    private readonly IShopsRepository _shopsRepository;
+    private readonly IItemManager _itemManager;
+    private readonly ILogger<ShopsManager> _logger;
+
+    public ShopsManager(IShopsRepository shopsRepository, IItemManager itemManager, ILogger<ShopsManager> logger)
+    {
+        _shopsRepository = shopsRepository;
+        _itemManager = itemManager;
+        _logger = logger;
+    }
+
+    public async Task<ShopItem[]> GetShopItems(uint npcVnum)
+    {
+        // get shops always from db not from cache
```

Comment:
instead of getting it from db, you can create a local hashmap that store them or an array and call db only once, as a temporary solution.

Reply:
Do u mean https://learn.microsoft.com/en-us/dotnet/api/system.collections.arraylist ?

Because that type is kinda outdated

Reply:
List<ShopItem> @MeikelLP